fe/fe-core/src/main/java/org/apache/doris/nereids/util/Utils.java

```diff
@@ -28,10 +28,7 @@ public class Utils {
      * @return quoted string
      */
     public static String quoteIfNeeded(String part) {
-        if (part.matches("[a-zA-Z0-9_]+") && !part.matches("\\d+")) {
-            return part;
-        } else {
-            return part.replace("`", "``");
-        }
+        return part.matches("\\w*[\\w&&[^\\d]]+\\w*")
```

Comment:
this pattern is not intuitional.  it is better add a comment to explain the pattern means all legal string except pure digit string.